provider/etcd/etcd.go

```diff
@@ -160,13 +160,23 @@ func (a *Alerts) Get(fp model.Fingerprint) (*types.Alert, error) {
 
 // Put adds the given alert to the set.
 func (a *Alerts) Put(alerts ...*types.Alert) error {
+	return a.put(false, alerts...)
+}
+
+func (a *Alerts) PutFromEtcd(alerts ...*types.Alert) error {
+	return a.put(true, alerts...)
+}
+
+func (a *Alerts) put(fromEtcd bool, alerts ...*types.Alert) error {
 
 	for _, alert := range alerts {
 		fp := alert.Fingerprint()
 
 		// Check that there's an alert existing within the store before
 		// trying to merge.
-		if old, err := a.alerts.Get(fp); err == nil {
+		var old *types.Alert
+		var err error
+		if old, err = a.alerts.Get(fp); err == nil {
```

Comment:
I needed to pass "old" in a function down below.  and I wasn't sure if line 179
`if old, err := `

would create a new old, because I'm not that familiar with golang.  I should have looked it up, but was in a hurry.  Do you know?

Reply:
:= should only create a new var if it doesn't exist. And it needs at least one new var to create (had to look that up). Not a big deal either way. I'm super new to golang as well and just wanted to learn why you did it